### PR TITLE
Create Trust provider packages

### LINKED ISSUE #5: Create Trust Provider package
Based on https://developer.trustwallet.com/trust-provider, it would be great to have a provider as a npm package:

`import trustProvider from "@trustwallet/provider"`

and implement simple methods to access:
```javascript
trustProvider.isEnabled: Bool
trustProvider.signTransaction(network: number, transaction: any): Promise<Result>
trustProvider.getAccounts(): Promise<[]Account>
```

### PR DESCRIPTION
**Pull request by leoneparise**
Fixes #5.

I needed it published before I test it here. @vikmeup please publish this packages as soon as possible.

### CODE REVIEW

File: packages/types/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+ssh://git@github.com/trustwallet/web-core.git"
+  },
+  "main": "",
+  "types": "index.d.ts",
+  "scripts": {},
+  "dependencies": {
+    "@types/mocha": "^5.2.6"
```
Comment (wpank) [inline packages/types/package.json]: Why is this a dependency for this types package?

File: packages/provider/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  ],
+  "devDependencies": {
+    "@babel/cli": "^7.1.2",
+    "@babel/core": "^7.1.2",
+    "@babel/node": "^7.0.0",
+    "@babel/polyfill": "^7.0.0",
+    "@babel/preset-env": "^7.1.0",
+    "@babel/preset-typescript": "^7.1.0",
+    "@babel/register": "^7.0.0",
+    "@types/chai": "^4.1.6",
```
Comment (wpank) [inline packages/provider/package.json]: I think using jest is better than mocha/chai. Desktop already uses jest for everything. I think it'd be good to use the testing env from that.

**Comment by wpank** [comment]
@leoneparise also can you fix the codacy formatting?

File: packages/provider/package.json (hunk 2 above)
Comment (leoneparise) [reply packages/provider/package.json]: Good catch. I'll remove it

**Comment by vikmeup** [comment]
@leoneparise I think we need merge first before I can publish it

File: packages/types/package.json (hunk 1 above)
Comment (leoneparise) [reply packages/types/package.json]: Removed!

**Pull request merged**